src/google.js

```diff
@@ -10,13 +10,147 @@
  * governing permissions and limitations under the License.
  */
 const { URL } = require('url');
+const { google } = require('googleapis');
 
 const { fetch } = require('@adobe/helix-fetch').context({
   httpsProtocols:
   /* istanbul ignore next */
   process.env.HELIX_FETCH_FORCE_HTTP1 ? ['http1'] : ['http2', 'http1'],
 });
 const { utils } = require('@adobe/helix-shared');
+const cache = require('./cache');
+
+/**
+ * Remember the access token for future action invocations.
+ */
+let tokenCache = {};
+
+function createOAuthClient(options, creds) {
+  const oAuth2Client = new google.auth.OAuth2(options);
+  oAuth2Client.setCredentials({
+    scope: 'https://www.googleapis.com/auth/drive.readonly',
+    ...tokenCache,
+    ...creds,
+  });
+  oAuth2Client.on('tokens', (tokens) => {
+    tokenCache = tokens;
+  });
+
+  return oAuth2Client;
+}
+
+const createCachedOAuthClient = cache(createOAuthClient);
+
+async function getUncachedIdFromPath(path, parentId, log, options) {
```

Comment:
why don't you move this resolution to data-embed, then you don't need any google api access here....

Reply:
It would require changing the data-embed API from `url` to `rootid, type, path` and add complexity there that is better kept here.